### DISCUSSION TITLE
Library is empty

### DISCUSSION THREAD

**Discussion by coachacg** [itsToggle/plex_debrid#502] [Q&A] [answered]
Hello All,

I have installed everything fresh except for Plex but updated to the latest version

Configuration:
-OS: MacOS Ventura 13.4.1
-Python 3.9.6
-Plex Version 1.32.4.7195-7c8f9d3b6
-"MacFUSE 4.5
-"rclone-darwin" v1.58.1-rd.2.2

-Both Plex and plex_debrid are running from the same host on the host OS
-"Plex server address": "http://localhost:32400/"
-"Plex "movies" library": "2",
-"Plex "shows" library": "2",

plex_debrid is throwing the following error:
[09/07/23 20:23:49] [plex error]: couldnt reach local plex library section '2' at server address: http://localhost:32400 - or this library really is empty.
[09/07/23 20:23:49] [plex error]: Your library seems empty. To prevent unwanted behaviour, no further downloads will be started. If your library really is empty, please add at least one media item manually.

Plex logs show connection attempts from 127.0.0.1:
Jul 09, 2023 20:37:52.245 [0x16ea37000] DEBUG - Request: [[::1]:57436 (Loopback)] GET /library/sections/ (2 live) #e1 GZIP Signed-in Token (*****) (Mac mini)
Jul 09, 2023 20:37:52.247 [0x16e28f000] DEBUG - Completed: [[::1]:57436] 200 GET /library/sections/ (2 live) #e1 GZIP 1ms 857 bytes (pipelined: 1)
Jul 09, 2023 20:37:52.250 [0x16ea37000] DEBUG - Request: [[::1]:57436 (Loopback)] GET /library/sections/1/all?type=1 (2 live) #e8 GZIP Signed-in Token (*****) (Mac mini)
Jul 09, 2023 20:37:52.251 [0x16ea37000] WARN - [Req#e8] Missing X-Plex-Container-Size header. This will fail with status code 400 in the future.
Jul 09, 2023 20:37:52.258 [0x16e31b000] DEBUG - Completed: [[::1]:57436] 200 GET /library/sections/1/all?type=1 (2 live) #e8 GZIP 7ms 4794 bytes (pipelined: 2)
Jul 09, 2023 20:37:52.260 [0x16ea37000] DEBUG - Request: [[::1]:57436 (Loopback)] GET /library/sections/2/all?type=2 (2 live) #e9 GZIP Signed-in Token (*****) (Mac mini)
Jul 09, 2023 20:37:52.261 [0x16ea37000] WARN - [Req#e9] Missing X-Plex-Container-Size header. This will fail with status code 400 in the future.
Jul 09, 2023 20:37:52.262 [0x16e28f000] DEBUG - Completed: [[::1]:57436] 200 GET /library/sections/2/all?type=2 (2 live) #e9 GZIP 1ms 601 bytes (pipelined: 3)
Jul 09, 2023 20:37:52.264 [0x16ea37000] DEBUG - Request: [[::1]:57436 (Loopback)] GET /library/sections/2/all?type=3 (2 live) #ea GZIP Signed-in Token (*****) (Mac mini)
Jul 09, 2023 20:37:52.265 [0x16ea37000] WARN - [Req#ea] Missing X-Plex-Container-Size header. This will fail with status code 400 in the future.
Jul 09, 2023 20:37:52.266 [0x16e31b000] DEBUG - Completed: [[::1]:57436] 200 GET /library/sections/2/all?type=3 (2 live) #ea GZIP 1ms 601 bytes (pipelined: 4)
Jul 09, 2023 20:37:52.268 [0x16ea37000] DEBUG - Request: [[::1]:57436 (Loopback)] GET /library/sections/2/all?type=4 (2 live) #eb GZIP Signed-in Token (*****) (Mac mini)
Jul 09, 2023 20:37:52.268 [0x16ea37000] WARN - [Req#eb] Missing X-Plex-Container-Size header. This will fail with status code 400 in the future.
Jul 09, 2023 20:37:52.270 [0x16e28f000] DEBUG - Completed: [[::1]:57436] 200 GET /library/sections/2/all?type=4 (2 live) #eb GZIP 1ms 601 bytes (pipelined: 5)

I changed debug printing to 'true' but no additional details. Any help is greatly appreciated. Thank you!

**Comment by coachacg** [comment]
I emptied my Watchlist in Plex and added one movie. Plex_Debrid noticed the changed but error remains and no downloads occur.

[09/07/23 20:48:27] [plex] item: "Guardians of the Galaxy Vol. 3" found in *****`s watchlist
[09/07/23 20:48:28] [plex] getting plex library section/s "Movies","TV Shows" ... done
[09/07/23 20:48:28] [plex error]: couldnt reach local plex library section '2' at server address: http://localhost:32400 - or this library really is empty.
[09/07/23 20:48:28] [plex error]: Your library seems empty. To prevent unwanted behaviour, no further downloads will be started. If your library really is empty, please add at least one media item manually.

**Comment by coachacg** [comment]
Here is another update. Adding movies manually to my torrents list in Real Debrid via a magnet link adds the movie to my Plex library for streaming and I see it on the remote drive. Adding movies to the Plex watchlist still has no affect other than it being acknowledged in the log. No scraping.

**Comment by coachacg** [comment, marked as answer]
For anyone having the same issue I was able to resolve it. It was actually something very stupid. My TV Shows library was empty causing everything to fail. I was able to figure this out by limiting the sources to update under Settings/Library Services/Library Collection Service/Edit/Edit Library Collection Service/Plex Library Check and adding the source of only a library with content. Once I added the empty library back the issue returned. Just make sure every library has at least one file.

**Reply by bbennett2** [reply]
just wanted to say thank you! spent a few hours today getting this installed, only to receive the same error.  Added a tv show to my tv folder and that fixed it. thanks so much for taking the time to post your solution!
